Sources/Macros/SampleBuilder/SampleBuilderMacro+Class.swift

```diff
@@ -0,0 +1,200 @@
+                }
+                
+                return nil
+            }.map {
+                ParameterItem(
+                    identifierName: $0.0,
+                    identifierType: $0.1,
+                    category: $0.2
+                )
+            }
+        }
+        
+        
+        let largestParameterList = initMember
+            .map {
+                getParametersFromInit(initSyntax: $0)
+            }.max {
+                $0.count < $1.count
+            } ?? []
+        
+        return largestParameterList
+    }
+    
+    static func getDataCategory(
+        from variableDecl: VariableDeclSyntax,
+        generatorType: DataGeneratorType,
+        context: MacroExpansionContext
+    ) -> DataCategory? {
+        guard let attributes = variableDecl.attributes
+            .first(where: {
+                $0.as(AttributeSyntax.self)?
+                    .attributeName
+                    .as(IdentifierTypeSyntax.self)?
+                    .name.text == "SampleBuilderItem"
+            })?.as(AttributeSyntax.self)
+        else {
+            return DataCategory.noCategory
+        }
+        
+        if generatorType == .default {
+            // Since default will always the return the same value,
+            // Using SampleBuilderItem is useless and will throw a warning.
+            SampleBuilderDiagnostic.report(
+                diagnostic: .sampleBuilderItemRedundant,
+                node: .init(attributes),
+                context: context)
+        }
+        
+        if let simpleCategoryString = attributes // All categories except image
+            .arguments?
+            .as(LabeledExprListSyntax.self)?
+            .first?.as(LabeledExprSyntax.self)?
+            .expression.as(MemberAccessExprSyntax.self)?
+            .declName.baseName.text {
+            
+            return DataCategory(rawValue: simpleCategoryString)
+        }
+        
+        if let imageCategoryExpression = attributes
+            .arguments?
+            .as(LabeledExprListSyntax.self)?
+            .first?.as(LabeledExprSyntax.self)?
+            .expression.as(FunctionCallExprSyntax.self),
+           
+            imageCategoryExpression
+            .calledExpression.as(MemberAccessExprSyntax.self)?
+            .declName.baseName.text == "image" {
+            
+            // Image's width and height
+            let argumentsValues = imageCategoryExpression.arguments.compactMap {
+                Int($0.expression.as(IntegerLiteralExprSyntax.self)?.literal.text ?? "")
+            }
+            
+            guard argumentsValues.count == 2
+            else {
+                return DataCategory.noCategory
+            }
+            
+            return DataCategory(imageWidth: argumentsValues[0], height: argumentsValues[1])
+        }
+        
+        return DataCategory.noCategory
+    }
+    
+    static func getParametersFromInit(
```

Comment:
Same here, This function is identical to the one used in structs. Let's remove this one and keep one single logic for both classes and structs.